examples/PyTorch/README.md

```diff
@@ -0,0 +1,98 @@
+# Using tensorhive for running distributed trainings in PyTorch
+
+This example is based on [this implementation](https://github.com/pytorch/examples/tree/master/dcgan)
+```
+git clone https://github.com/pytorch/examples.git
+```
+
+## Downloading the dataset
+
+You can download the LSUN dataset by cloning [this repo](https://github.com/fyu/lsun) 
+
+```
+git cllone https://github.com/fyu/lsun.git
+```
+and running
+```
+python download.py -c bedroom
+```
+
+For testing and/or benchmarking purposes it is also possible to the train the model on fake dataset. To do this run this command instead of the one given in the Usage section:
+
+```
+CUDA_VISIBLE_DEVICES=0 python main.py --init-method tcp://127.0.0.1:20011 --rank 0 --world 2 --dataset fake --cuda
+CUDA_VISIBLE_DEVICES=1 python main.py --init-method tcp://127.0.0.1:20011 --rank 1 --world 2 --dataset fake --cuda
+```
+
+## Making the code parallel
+
+It is actually very easy to allow PyTorch code to run on multiple GPUs
+
+### Locally
+
+All you need to do to make a data parallel model that can be run on multiple GPUs locally:
+```
+from torch import nn
+<your code including creating the model>
+model = nn.DataParallel(model)
+```
+That's it. After that you just control which GPUs you would like your code to run on by specifing their number in CUDA_VISIBLE_DEVICES before your command (tensorhive takes care of that for you when you specify the devices while creating the task).
+
+### Distributed
+
+To make the code distributed apply provided patch by running:
+
+```
+git apply distributed.patch
+```
+
+You might want to checkout commit 0c1654d6913f77f09c0505fb284d977d89c17c1a to make sure there are no conflicts before applying the patch.
+
+To learn how to easily adapt PyTorch code for distributed learning please examine the patch. 
+
+## Usage
+```
+# set accroding to your configuration
+export GLOO_SOCKET_IFNAME=eth0 
+```
+
+```
+CUDA_VISIBLE_DEVICES=0 python main.py --init-method tcp://127.0.0.1:20011 --rank 0 --world 2 --dataset lsun --dataroot <path to lsun dataset> --cuda
+CUDA_VISIBLE_DEVICES=1 python main.py --init-method tcp://127.0.0.1:20011 --rank 1 --world 2 --dataset lsun --dataroot <path to lsun dataset> --cuda
+
+For more details about the usage refer to the [original repo](https://github.com/pytorch/examples/tree/master/dcgan)
+```
+
```

Comment:
The README to this line is a copy of the benchmark README in the dnn_training_benchmarks repo. I think we should provide a link to that README here instead of introducing such redundancy. So, instructions for running the benchmark and results on chosen hardware should be there, while instructions, examples of running it using TH and comparison with no-TH version should be here.

Reply:
I think that this redundancy is pretty useful for the user. Having to open another repo just for the instructions for running (that also has links to other repos) seems like a unneeded hassle. Of course after fixing everything in the dnn_training_benchmarks repo I will copy the changes over here too

Reply:
OK

Reply:
Hmm, this redundancy is starting to generate problems. At least because the distributed patch is also doubled - you updated it and I got confused, using the old one from this repo. Maybe we should at least keep only one patch in the dnn_training_benchmarks repo, and add instructions for downloading the patch here in the README?

Reply:
I think the problem is only on our side during this checking but since it makes it easier to use by the user i think it's worth it (I updated the patch here to the newest and probably final version from ddn_benchmarks)

Reply:
Then how is it that we still have "git cllone" here?

Reply:
The improvements made in the dnn_training_benchmarks repo are still not reflected here. Maybe consider once again only referencing the example here, instead of copying the content.

Reply:
done